character-sfx.js

```diff
@@ -259,6 +284,20 @@ class CharacterSfx {
   
     };
     _handleNarutoRun();
+    
+
+    const _handleGasp = () =>{
+      if(timeSeconds - this.lastRunningTime > 2 && !this.player.avatar.narutoRunState && timeSeconds - this.narutoRunFinishTime > 0.5){
+        this.runStep = 0;
+      }
+      // if running step is more than 10
+      if(currentSpeed < 0.1 && timeSeconds - this.lastWalkTime > 0.01 && (this.runStep > 10 || this.willGasp) && !this.player.avatar.narutoRunState){

```

Comment:
I don't think we should count run steps at all.

We should simply set `willGasp` after a certain amount of time.

Reply:
That would be more consistent with the naruto run for example, and it would make the gasp not dependent on animation (which is weird).

Reply:
remove `runStep` and just use `willGasp` now: https://github.com/webaverse/app/pull/2835/commits/750d3d0a8e4cf7e31d127fb389f8f282c6d649d6

Reply:
remove `runStep` and play the grunt by `willGasp`: https://github.com/webaverse/app/pull/2835/commits/5cba3697b2848b23ddce62fd62b7632206f4e9a1